packages/wrangler/src/dev.tsx

```diff
@@ -1003,3 +1042,26 @@ export function getBindings(
 
 	return bindings;
 }
+
+/**
+ * Merge two arrays of objects, overriding elements in `source` with
+ * elements from `override`, if they have the same `key` value
+ */
+function mergeWithOverride<T>(
+	source: Array<T>,
+	override: Array<T>,
+	key: keyof T
+): Array<T> {
+	const sourceMap: Map<T[typeof key], T> = new Map();
+	source.forEach((el) => sourceMap.set(el[key], el));
+
+	// preserve everything in `override` as it takes precedence over `source`,
+	// and add remaining elements from `source` that were not overriden
+	override.forEach((el) => {
+		if (sourceMap.has(el[key])) {
+			sourceMap.delete(el[key]);
+		}
+	});
+
+	return [...override, ...sourceMap.values()];
```

Comment:
This seems a bit of a complicated approach. Would the following also work?

```suggestion
	override.forEach((el) => sourceMap.set(el[key], el));
	return Array.from(sourceMap.values());
```

Reply:
oh ofc, because Map keys are unique. Completely overlooked that 😓 . Fixed!